src/pretix/static/pretixpresale/js/ui/cookieconsent.js

```diff
@@ -0,0 +1,103 @@
+/*global $ */
+
+$(function () {
+    function _send_event(type) {
+        // Event() is not supported by IE11
+        var e = document.createEvent('Event')
+        e.initEvent(type, true, true)
+        document.dispatchEvent(e)
+    }
+
+    if ($("#cookie-consent-storage-key").length) {
+        window.__pretix_cookie_consent = {}
+    } else {
+        window.__pretix_cookie_consent = null
+        return
+    }
+
+    var storage_key = $("#cookie-consent-storage-key").text()
+    var storage_val = window.localStorage[storage_key]
+    var show_dialog = false
+    if (!storage_val) {
+        show_dialog = true
+        storage_val = {}
+
+        if ($("#cookie-consent-from-widget").length) {
+            var consented = JSON.parse($("#cookie-consent-from-widget").text())
+            $("#cookie-consent-details input[type=checkbox][name]").each(function () {
+                storage_val[$(this).attr("name")] = consented.indexOf($(this).attr("name")) > -1
+                $(this).prop("checked", consented.indexOf($(this).attr("name")) > -1)
+            })
+            window.localStorage[storage_key] = JSON.stringify(storage_val)
+            show_dialog = false
+        }
+    } else {
+        storage_val = JSON.parse(storage_val)
+        $("#cookie-consent-details input[type=checkbox][name]").each(function () {
+            if (typeof storage_val[$(this).attr("name")] === "undefined") {
+                // A new cookie type has been added that we haven't asked for yet
+                show_dialog = true
+            } else if (storage_val[$(this).attr("name")]) {
+                $(this).prop("checked", true)
+            }
+        })
```

Comment:
A very rare edge-case: in an earlier session, I set the consent through the widget, then continue to pretix, the consent is saved to localStorage. Later I get back to the website with the widget embedded, but in the meantime another provider to the cookie consent was added. The widget triggers a new cookie-consent-dialog (due to the new provider) and – as I navigate to pretix – transfers its consent over to pretix through the `#cookie-consent-from-widget`. But this time, as a cookie-consent already exists for pretix from a previous session, but without the newly added provider, I get the cookie dialog that I answered in the widget again.

Reply:
Well… yes. But in 90% of cases the widget runs on a different domain and gets an isolated localStorage by modern browsers, so it's even more rare.